source/index.ts

```diff
@@ -43,13 +43,13 @@ const checkValueType = <Key = unknown>(key: string, value: Key): void => {
 const INTERNAL_KEY = '__internal__';
 const MIGRATION_KEY = `${INTERNAL_KEY}.migrations.version`;
 
-export class Conf<T extends any = any> implements Iterable<[keyof T, T[keyof T]]> {
+export class Conf<T extends { [key: string]: any } = any> implements Iterable<[keyof T, T[keyof T]]> {
```

Comment:
Shouldn't it be:

```suggestion
export class Conf<T extends {[key: string]: any} = {[key: string]: any}> implements Iterable<[keyof T, T[keyof T]]> {
```

?

Reply:
Yes, I initially had some issues with that. But it seems to be working now.